src/config/config_collapse.c

```diff
@@ -121,14 +117,10 @@ __config_merge_scan(
             WT_ERR_MSG(session, EINVAL, "Invalid configuration key found: '%s'", k.str);
 
         /* Include the quotes around string keys/values. */
-        if (k.type == WT_CONFIG_ITEM_STRING) {
-            --k.str;
-            k.len += 2;
-        }
-        if (v.type == WT_CONFIG_ITEM_STRING) {
-            --v.str;
-            v.len += 2;
-        }
+        if (k.type == WT_CONFIG_ITEM_STRING)
+            WT_CONFIG_PRESERVE_QUOTES(&k);
```

Comment:
This looks a bit suspicious to me. What if we are handling the first key in the config string here? `WT_CONFIG_PRESERVE_QUOTES` will try to access `str[-1]` which is before the first symbol in the config string. There's no problem with the handling of values below and in `tiered_handle.c`. Do we really need to support quoted keys?

Reply:
Nice catch!  I think we do have to handle this case - as a matter of fact the code it replaces expects every key and value to be quoted (it doesn't even check).  This code is not the case that is failing on the ticket (that's `tiered_handle.c`).  I'm going to code it in a safer way, swapping the order of checks.